### ISSUE TITLE
Setting RNG state while looping through model generate, Reproducibility.

### ISSUE THREAD

**Issue by ArtHughes** [karpathy/nanoGPT#426]
Hi There. I am looping though the model generate within sample.py to create a temperature profile of the model. I am looping through a range of min and max temperatures. This works fine. However, if I change the min and max parameters, I get different results. Should I re-seed the torch rngs to a constant (1337) whenever I call model generate within the loop, or save/restore the state of the rngs, around the call to model generate, or is this expected behavior? I've already set my CUDA and CBLAS libs to be deterministic. Thanks.

**Comment by alxndrTL** [comment]
In this snippet :
```
with torch.no_grad():
    with ctx:
        for k in range(num_samples):
            y = model.generate(x, max_new_tokens, temperature=temperature, top_k=top_k)
            print(decode(y[0].tolist()))
            print('---------------')
```
if you want all samples to be the same, yes, you would have to reset the RNG seed before calling `model.generate` by using :

```
torch.manual_seed(seed)
torch.cuda.manual_seed(seed)
```

I hope this answers your question!

**Comment by ArtHughes** [comment]
Thanks so much for the quick response. That's perfect. I'll reset the seed within the loop.
